File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/repository/AuthRepository.kt
package com.ssafy.ssabree.core.repository

import com.ssafy.ssabree.core.repository.model.SignUpInfo

interface AuthRepository {
    suspend fun login(email: String, password: String): Result<Unit>
    suspend fun refreshTokens(): Result<Unit>
    suspend fun logout(): Result<Unit>

    suspend fun isEmailUsed(email: String): Result<Boolean>
    suspend fun signUp(req: SignUpInfo): Result<Unit>

    suspend fun requestSsafyVerification(mattermostId: String, generation: Int, name: String): Result<Unit>
    suspend fun confirmSsafyVerification(mattermostId: String, code: String): Result<Unit>

    suspend fun requestPasswordResetCode(destination: String): Result<Unit>
    suspend fun confirmPasswordReset(code: String, newPassword: String): Result<Unit>

    suspend fun findId(mattermostId: String): Result<String>
    suspend fun resetPassword(mattermostId: String, newPassword: String): Result<Unit>

    suspend fun withdraw(): Result<Unit>
}

File: iOS/ssabree/ssabree/Core/Repository/AuthRepository.swift
import Foundation

protocol AuthRepository {
    func login(email: String, password: String) async -> Result<Void, Error>
    func refreshTokens() async -> Result<Void, Error>
    func logout() async -> Result<Void, Error>
    func isLoggedIn() -> Bool

    func isEmailUsed(email: String) async -> Result<Bool, Error>
    func signUp(info: SignUpInfo) async -> Result<Void, Error>

    func requestSsafyVerification(mattermostId: String, generation: Int, name: String) async -> Result<Void, Error>
    func confirmSsafyVerification(mattermostId: String, code: String) async -> Result<Void, Error>

    func findId(mattermostId: String) async -> Result<String, Error>
    func resetPassword(mattermostId: String, newPassword: String) async -> Result<Void, Error>

    func withdraw() async -> Result<Void, Error>

    func getMyMemberId() async -> Result<Int, Error>
}
